src/main/java/com/palonskiy/serice/AuthorServiceImpl.java

```diff
@@ -50,88 +39,26 @@ public void update(AuthorDto authorDto) {
-                    bookService.addOnlyBook(bookAuthorDto.getBook());
-                    bookId = bookService.getId();
-                }
-                bookAuthorService.add(bookAuthorDto, authorId, bookId);
-            } else throw new ExistingEntityException();
-        } else
-            throw new NullAuthorException();
-    }
-
-    @Override
-    public void addOnlyAuthor(AuthorDto authorDto) {
-        authorDao.add(AuthorConverter.toAuthor(authorDto));
-    }
-
-    @Override
-    public Long getId() {
-        return authorDao.getId();
+    public Author add(AuthorDto authorDto) {
+        return authorDao.add(AuthorConverter.toAuthor(authorDto));
     }
 
     @Override
     public void delete(int id) {
-        List<Long> books = bookAuthorService.findBooksByAuthor(Long.valueOf(id));
-        for (Long bookId : books) {
-            if (bookAuthorService.findAuthorsByBook(bookId).size() < 2) {
-                bookService.delete(Math.toIntExact(bookId));
-            }
-        }
-        authorDao.delete(Long.valueOf(id));
-    }
-
-    @Override
-    public List<BookAuthorDto> sortByBirthday(int year) {
-        return Optional.of(bookAuthorService.get().stream()
-                .filter(bookAuthor -> (bookAuthor.getAuthor().getBirthday().getYear() == year))
-                .collect(Collectors.toList()))
-                .filter(a -> !a.isEmpty())
-                .orElseThrow(() -> new NoResultException());
-    }
-
-    @Override
-    public List<BookAuthorDto> sortBySex(String sex) {
-        return Optional.of(bookAuthorService.get().stream()
-                .filter(bookAuthor -> (bookAuthor.getAuthor().getSex().toString().equals(sex)))
-                .collect(Collectors.toList()))
-                .filter(a -> !a.isEmpty())
-                .orElseThrow(() -> new NoResultException());
+        authorDao.delete(authorDao.getById(Long.valueOf(id)));
     }
 
     @Override
-    public List<BookAuthorDto> sortByName(String name) {
-        return Optional.of(bookAuthorService.get().stream()
-                .filter(bookAuthor -> (bookAuthor.getAuthor().getFirstName().startsWith(name)))
-                .collect(Collectors.toList()))
-                .filter(a -> !a.isEmpty())
-                .orElseThrow(() -> new NoResultException());
-    }
-
-    @Override
-    public List<BookAuthorDto> sortBySecondName(String name) {
-        return Optional.of(bookAuthorService.get().stream()
-                .filter(bookAuthor -> (bookAuthor.getAuthor().getSecondName().startsWith(name)))
-                .collect(Collectors.toList()))
-                .filter(a -> !a.isEmpty())
-                .orElseThrow(() -> new NoResultException());
+    public List<BookDto> getByJoinField(Object obj, String fieldName) {
+        List<BookDto> list = BookConverter.toDtoList(authorDao.getByJoinField(obj, fieldName));
+        if (list.isEmpty()) {
+            throw new NoResultException();
```

Comment:
Bad decision to throw an exception here. Return empty list.
It is common good practice to return empty list if you requested some list of object (search or getall or any other calls)

Reply:
Have deleted if block